resources/assets/js/topics.js

```diff
@@ -0,0 +1,72 @@
+let topics = null;
+
+export default {
+    all() {
+        if (topics !== null) {
+            return new Promise(function (resolve) {
+                resolve(topics);
+            });
+        }
+
+        return new Promise(function (resolve, reject) {
+            Vue.http.get('topics').then(
+                response => {
+                    topics = response.data;
+                    resolve(topics);
+                },
+                response => {
+                    reject();
+                    console.log('error', response);
+                }
+            );
+        });
+    },
+
+    add(topic) {
+        return new Promise( (resolve, reject) => {
+            Vue.http.post('topics', {title: topic.title, description: topic.description})
+                .then(
+                    (response) => {
+                        topics.push(response.data);
+                        resolve(response);
+                    },
+                    (response) => {
+                        reject(response);
+                        console.log('error', response);
+                    }
+                );
+        });
+
+    },
+
+    flag(topic, status) {
+        Vue.http.patch('topics/' + topic.id, {status: status})
+            .then(
+                () => {
```

Comment:
Inconsistent params on your callbacks here is confusing. Some have parens, some don't, some specify response, some don't. let's normalize please :)